inference-engine/tests/functional/shared_test_classes/src/single_layer/activation.cpp

```diff
@@ -163,70 +170,26 @@ ngraph::ParameterVector ActivationParamLayerTest::createActivationParams(ngraph:
-            inputs.push_back(blobLeakySlope);
-            break;
-        }
-        case ngraph::helpers::ActivationTypes::HardSigmoid: {
-            auto blobHardSigmoidAlpha = inferRequest.GetBlob("alpha");
-            auto blobHardSigmoidBeta = inferRequest.GetBlob("beta");
-            float alpha = constantsValue[0], beta = constantsValue[1];
-            blobHardSigmoidAlpha = FuncTestUtils::createAndFillBlobWithFloatArray(blobHardSigmoidAlpha->getTensorDesc(), &alpha, 1);
-            blobHardSigmoidBeta = FuncTestUtils::createAndFillBlobWithFloatArray(blobHardSigmoidBeta->getTensorDesc(), &beta, 1);
-            inferRequest.SetBlob("alpha", blobHardSigmoidAlpha);
-            inferRequest.SetBlob("beta", blobHardSigmoidBeta);
-            inputs.push_back(blobHardSigmoidAlpha);
-            inputs.push_back(blobHardSigmoidBeta);
-            break;
-        }
-        case ngraph::helpers::ActivationTypes::Selu: {
-            auto blobHardSigmoidAlpha = inferRequest.GetBlob("alpha");
-            auto blobHardSigmoidLambda = inferRequest.GetBlob("lambda");
-            float alpha = constantsValue[0], lambda = constantsValue[1];
-            blobHardSigmoidAlpha = FuncTestUtils::createAndFillBlobWithFloatArray(blobHardSigmoidAlpha->getTensorDesc(), &alpha, 1);
-            blobHardSigmoidLambda = FuncTestUtils::createAndFillBlobWithFloatArray(blobHardSigmoidLambda->getTensorDesc(), &lambda, 1);
-            inferRequest.SetBlob("alpha", blobHardSigmoidAlpha);
-            inferRequest.SetBlob("lambda", blobHardSigmoidLambda);
-            inputs.push_back(blobHardSigmoidAlpha);
-            inputs.push_back(blobHardSigmoidLambda);
-            break;
-        }
-        default:
-            IE_THROW() << "Unsupported activation type for Params test type";
-    }
-}
-
-void ActivationParamLayerTest::Infer() {
-    inferRequest = executableNetwork.CreateInferRequest();
-
-    auto blobInput = inferRequest.GetBlob("Input");
-    blobInput = FuncTestUtils::createAndFillBlobFloat(blobInput->getTensorDesc());
-    inferRequest.SetBlob("Input", blobInput);
-    inputs.push_back(blobInput);
-
-    generateActivationBlob(constantsValue);
-
-    if (configuration.count(InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED) &&
-        configuration.count(InferenceEngine::PluginConfigParams::YES)) {
-        auto batchSize = executableNetwork.GetInputsInfo().begin()->second->getTensorDesc().getDims()[0] / 2;
-        inferRequest.SetBatch(batchSize);
+InferenceEngine::Blob::Ptr ActivationParamLayerTest::GenerateInput(const InferenceEngine::InputInfo &info) const {
+    InferenceEngine::Blob::Ptr blobPtr;
+    const std::string& name = info.name();
+    if (name == "negativeSlope") {
+        const auto elemnts_count = ngraph::shape_size(function->get_parameters()[1]->get_shape());
+        std::vector<float> param_data(elemnts_count);
+        std::iota(param_data.begin(), param_data.end(), 2);
```

Comment:
It's a good idea to use unique slope values, but it is probably better to use the same generation rule in both ActivationLayerTest and ActivationParamLayerTest. Just to be a bit more consistent in case someone would compare the outcome of smoke_Activation_Basic_Prelu_Const and smoke_Activation_Basic_Prelu_Param.

Reply:
Right, done (changed to start from -10 in both cases). Thank you!